### PR TITLE
feat: add Interceptor to record payload info

### PR DESCRIPTION
**Pull request by pangdayuan1**

### CODE REVIEW

File: arex-storage-web-api/src/main/java/com/arextest/storage/interceptor/RequestResponseMetricInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    String requestURI = request.getRequestURI();
+    long requestBodySize = request.getContentLengthLong();
+    String appId = null;
+    String recordId = null;
+    String category = null;
+    if (request instanceof ContentCachingRequestWrapper) {
+      byte[] contentAsByteArray = ((ContentCachingRequestWrapper) request).getContentAsByteArray();
+      String requestBody = new String(contentAsByteArray);
+      if (StringUtils.isNotEmpty(requestBody)) {
+        JsonNode jsonNode = objectMapper.readTree(requestBody);
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/interceptor/RequestResponseMetricInterceptor.java:41]: Parsing the request body is not recommended, metric tags can be transmitted via the request header

File: arex-storage-web-api/src/main/java/com/arextest/storage/metric/MetricListener.java [hunk 2]
```diff
@@ -17,4 +17,9 @@ public interface MetricListener {
    * record match strategy count with tags.
    */
   void recordMatchingCount(String metricName, Map<String, String> tags);
+
+  /**
+   * record request and response size with tags.
+   */
+  void recordRequestAndResponseSize(String metricName, Map<String, String> tags, long requestSize, long responseSize);
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/metric/MetricListener.java:24]: Here you need a recordSize method

`void recordSize(String metricName, Map<String, String> tags, long timeMillis);`

File: arex-storage-web-api/src/main/java/com/arextest/storage/metric/RequestResponseMetricService.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+import java.util.Map;
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.collections4.CollectionUtils;
+
+/**
+ * Records the request and response size with the appropriate MetricListeners.
+ * created by xinyuan_wang on 2023/12/25
+ */
+@Slf4j
+public class RequestResponseMetricService {
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/metric/RequestResponseMetricService.java:14]: The metric-related logic should be placed in the `MetricService` class.

File: arex-storage-web-api/src/main/java/com/arextest/storage/interceptor/RequestResponseMetricInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+import org.springframework.web.util.ContentCachingResponseWrapper;
+
+/**
+ * record request and response interceptor
+ * created by xinyuan_wang on 2023/12/25
+ */
+@Component
+@Slf4j
+public class RequestResponseMetricInterceptor implements HandlerInterceptor {
+  private static final String APP_ID_HEADER = "app-id";
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/interceptor/RequestResponseMetricInterceptor.java:20]: Suggest rename to `client-app`

File: arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java [hunk 5]
```diff
@@ -33,11 +34,18 @@ public class AgentWorkingMetricService {
   private static final String SCENE = "scene";
   private static final String SCENE_RECORD = "record";
   private static final String SCENE_REPLAY = "replay";
+  private static final String ENTRY_PAYLOAD_NAME = "service.entry.payload";
+  private static final String RECORD_ID = "recordId";
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java:38]: Please remove field `RECORD_ID`

**Comment by mr3** [comment]
Please fix [5 New issues]

File: arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java [hunk 6]
```diff
@@ -33,11 +34,17 @@ public class AgentWorkingMetricService {
   private static final String SCENE = "scene";
   private static final String SCENE_RECORD = "record";
   private static final String SCENE_REPLAY = "replay";
+  private static final String ENTRY_PAYLOAD_NAME = "service.entry.payload";
+  private static final String REQUEST_URI = "requestURI";
```
Comment (mr3) [inline arex-storage-web-api/src/main/java/com/arextest/storage/metric/AgentWorkingMetricService.java:38]: Sugguest rename to `path`

**Review by mr3 — APPROVED** [review]

**Review by srotony — APPROVED** [review]

**Pull request merged**